homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -276,6 +277,46 @@ def async_get_pipelines(hass: HomeAssistant) -> Iterable[Pipeline]:
     return pipeline_data.pipeline_store.data.values()
 
 
+async def async_update_pipeline(
+    hass: HomeAssistant,
+    pipeline: Pipeline,
+    *,
+    conversation_engine: str | UndefinedType = UNDEFINED,
+    conversation_language: str | UndefinedType = UNDEFINED,
+    language: str | UndefinedType = UNDEFINED,
+    name: str | UndefinedType = UNDEFINED,
+    stt_engine: str | None | UndefinedType = UNDEFINED,
+    stt_language: str | None | UndefinedType = UNDEFINED,
+    tts_engine: str | None | UndefinedType = UNDEFINED,
+    tts_language: str | None | UndefinedType = UNDEFINED,
+    tts_voice: str | None | UndefinedType = UNDEFINED,
+    wake_word_entity: str | None | UndefinedType = UNDEFINED,
+    wake_word_id: str | None | UndefinedType = UNDEFINED,
+) -> None:
+    """Update a pipeline."""
+    pipeline_data: PipelineData = hass.data[DOMAIN]
+
+    updates: dict[str, Any] = pipeline.to_json()
+    updates.pop("id")
+    for key, val in (
+        ("conversation_engine", conversation_engine),
+        ("conversation_language", conversation_language),
+        ("language", language),
+        ("name", name),
+        ("stt_engine", stt_engine),
+        ("stt_language", stt_language),
+        ("tts_engine", tts_engine),
+        ("tts_language", tts_language),
+        ("tts_voice", tts_voice),
+        ("wake_word_entity", wake_word_entity),
+        ("wake_word_id", wake_word_id),
+    ):
```

Comment:
This really could make use of https://peps.python.org/pep-0692/, maybe add a comment here to refactor once we have bumped to Python 3.12?

Reply:
```suggestion
    # Refactor this once we bump to Python 3.12 and have https://peps.python.org/pep-0692/
    for key, val in (
        ("conversation_engine", conversation_engine),
        ("conversation_language", conversation_language),
        ("language", language),
        ("name", name),
        ("stt_engine", stt_engine),
        ("stt_language", stt_language),
        ("tts_engine", tts_engine),
        ("tts_language", tts_language),
        ("tts_voice", tts_voice),
        ("wake_word_entity", wake_word_entity),
        ("wake_word_id", wake_word_id),
    ):
```